notebooks/2019-11-12_Calendars_and_monthly_data.ipynb

```diff
@@ -0,0 +1,576 @@
+      "       4.36339797e-05, 3.60268132e-05, 3.00881638e-05, 4.35306283e-05,\n",
+      "       5.61130400e-05, 7.42721750e-05, 6.80996891e-05, 3.75807103e-05,\n",
+      "       3.38666614e-05, 3.41925661e-05, 2.94617275e-05, 3.45140616e-05,\n",
+      "       3.81903556e-05, 4.68480248e-05, 2.72760735e-05, 3.36759804e-05,\n",
+      "       5.02384646e-05, 8.06452663e-05, 6.21316722e-05, 4.41266056e-05,\n",
+      "       3.04287478e-05, 2.62973190e-05, 2.64842383e-05, 3.26729569e-05,\n",
+      "       4.52012209e-05, 4.85079363e-05, 3.67452449e-05, 3.40536790e-05,\n",
+      "       3.67230241e-05, 7.03531623e-05, 6.69856599e-05, 4.26577899e-05,\n",
+      "       3.21605548e-05, 2.25631738e-05, 3.23108943e-05, 2.39344427e-05],\n",
+      "      dtype=float32)\n",
+      "Coordinates:\n",
+      "  * time      (time) object 2274-12-16 00:00:00 ... 2299-11-16 00:00:00\n",
+      "    model_id  <U10 'HadGEM2-ES' \n",
+      " <xarray.DataArray 'pr' (time: 2412)>\n",
+      "array([2.7216340e-05, 2.1928223e-05, 2.5535373e-05, ..., 4.2512460e-05,\n",
+      "       4.3559990e-05, 4.7103615e-05], dtype=float32)\n",
+      "Coordinates:\n",
+      "  * time      (time) object 2100-01-16 12:00:00 ... 2300-12-16 12:00:00\n",
+      "    model_id  <U10 'bcc-csm1-1'\n"
+     ]
+    }
+   ],
+   "source": [
+    "for ii in range(len(var_list)):\n",
+    "    var_list[ii] = var_list[ii].sel(lat=[0., 90.],method='nearest').mean(dim=['lon','lat'])\n",
+    "print(var_list[0],'\\n',var_list[1])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Align the times\n",
+    "Finally, we can deal with the time coordinates. What we'll do is put all the dates on a 360 day calendar for the middle of the month. Since all months have 30 days in a 360 day calendar, the mid-month mark is on 16th 00:00:00. There is no automatic way to change from one calendar to another because the way you want to do it depends on why you are doing it. Here we are going to get the year and month for each date and put them as argument of the `cftime.Datetime360Day()` function which sets the date for the 360 day calendar."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def to_360day_monthly(da):\n",
+    "    '''Takes a DataArray. Change the \n",
+    "    calendar to 360_day and precision to monthly.'''\n",
+    "    val = da.copy()\n",
+    "    time1 = da.time.copy()\n",
+    "    for itime in range(val.sizes['time']):\n",
+    "        bb = val.time.values[itime].timetuple()\n",
+    "        time1.values[itime] = cftime.Datetime360Day(bb[0],bb[1],16)\n",
+    "\n",
+    "    # We rename the time dimension and coordinate to time360 to make it clear it isn't \n",
+    "    # the original time coordinate.\n",
+    "    val = val.rename({'time':'time360'})\n",
+    "    time1 = time1.rename({'time':'time360'})\n",
```

Comment:
I'm not sure renaming the time dataarray is needed, I think it will use the name given when added as a coordinate

Reply:
If I comment:
#time1 = time1.rename({'time':'time360'})
I get:
ValueError: cannot add coordinates with new dimensions to a DataArray

Reply:
oh right, yeah cause it's a coordinate dimension, don't mind me :)

Reply:
It's the same if I comment the renaming for val. I always have issues with assign_coords() so it's possible there is something I don't grasp here.

Reply:
I've looked a bit into it. It's possible to do 
`    # We rename the time dimension and coordinate to time360 to make it clear it isn't 
    # the original time coordinate.
    #val = val.rename({'time':'time360'})
    #time1 = time1.rename({'time':'time360'})
    val = val.assign_coords({'time360':time1})
    val = val.swap_dims({'time':'time360'})`
In this case, we keep both time coordinates (original and new). Which can be useful or can be a source for mistakes. I'll add it to the blog.